Make Graphviz engine configurable

markuspf opened the pull request:
I recently found myself wanting to use a different GraphViz renderer for `Splash`, so I added this option. Might be useful for others.

wilfwilson commented on doc/display.xml:
```diff
@@ -69,6 +69,17 @@
+        <Item>
+          this option can be used to specify the GraphViz engine to use
+          to render a <C>dot</C>. The valid choices are <C>"dot"</C>,
+          <C>"neato"</C>,<C>"circo"</C>, <C>"twopi"</C>, <C>"fdp"</C>,
```
There should be a space after the first comma.

wilfwilson commented on tst/standard/display.tst:
```diff
@@ -102,6 +102,9 @@ gap> dot{[1 .. 50]};
 #gap> Splash("string", rec(viewer := "xpdf"));
 #Error, Digraphs: Splash: usage,
 #the option <type> must be "dot" or "latex",
+#gap> Splash("string", rec(type := "dot", engine := "dott");
```
You're missing a closing bracket from your "test"

wilfwilson commented on doc/display.xml:
```diff
@@ -69,6 +69,17 @@
+        <Mark>engine</Mark>
+        <Item>
+          this option can be used to specify the GraphViz engine to use
+          to render a <C>dot</C>. The valid choices are <C>"dot"</C>,
```
`<C>dot</C>` -> `<C>dot</C> file` or `<C>dot</C> document`.

wilfwilson commented on tst/standard/display.tst:
```diff
@@ -102,6 +102,9 @@ gap> dot{[1 .. 50]};
 #the option <type> must be "dot" or "latex",
+#gap> Splash("string", rec(type := "dot", engine := "dott");
+#Error, Digraphs: Splash: usage,
+#the option <engine> must be "dot", "neato","twopi", "circo", "fdp", "sfdp", or "patchwork"
```
There's a missing space between neato and twopi.

markuspf commented:
I think I addressed your remarks.

Can you clarify what you mean by "some of the comments don't seem to be useful with the examples I've tried so far"?

markuspf commented:
Ah, its supposed to mean "commands", excuse my slowness.

According to its documentation `fdp`, is for drawing "undirected graphs":

```
fdp draws undirected graphs using a ``spring'' model. It relies on a force‐directed approach in the spirit of Fruchterman and  Reingold  (cf.  Soft‐
       ware‐Practice & Experience 21(11), 1991, pp. 1129‐1164).
```

Of course we could make splash more complicated to catch misuse; that'd be quite a lot of work for very little gain, or we could include bits of the graphviz documentation in the documentation of splash...

wilfwilson commented:
Oh yes, I meant 'commands'. I didn't spot the typo until just now either. I don't think we need to make anything more complicated.
